Update version compatibility

pfefferle opened the pull request:
## Proposed changes:
* Add PHP 8.5 to the test matrix to ensure compatibility with upcoming PHP version
* Update WordPress tested up to version 6.9

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Check that the GitHub Actions workflow runs successfully with PHP 8.5
* Verify the readme.txt reflects the correct WordPress version compatibility

### Changelog entry

- [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

- [ ] Patch
- [x] Minor
- [ ] Major

#### Type

- [ ] Added - for new features
- [x] Changed - for changes in existing functionality
- [ ] Deprecated - for soon-to-be removed features
- [ ] Removed - for now removed features
- [ ] Fixed - for any bug fixes
- [ ] Security - in case of vulnerabilities

#### Message

Update compatibility testing for PHP 8.5 and WordPress 6.9

</details>

pfefferle committed:
Update version compatibility

Add PHP 8.5 to test matrix and update WordPress tested up to version 6.9.

pfefferle committed:
Update PHP versions matrix in PHPUnit workflow

Removed PHP 8.5 from the main matrix and added it to the trunk WordPress version. This ensures tests run with the latest PHP version only on trunk, aligning with compatibility testing strategy.

matticbot committed:
Add changelog

obenland commented on .github/workflows/phpunit.yml:
```suggestion
        php-versions: ['7.2', '8.3', '8.5']
```
Maybe this could be an opportunity to cut down on the amount of parallel tests we run on every push?

* 7.2 is our minimum supported version currently.
* 8.3 is [the recommended version](https://wordpress.org/about/requirements/) by WordPress.
* 8.5 the latest.

pfefferle committed:
Merge branch 'trunk' into update/version-compatibility

pfefferle committed:
Update PHP versions in PHPUnit workflow

The GitHub Actions workflow for PHPUnit now tests against PHP 7.2, 8.3, and 8.5 instead of the previous broader range. This streamlines CI to focus on currently relevant PHP versions.

props @obenland
